src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```
@@ -51,14 +52,19 @@ public function __construct($registry, $classOrAlias, $property = null, $manager
     public function loadUserByUsername($username)
     {
         $repository = $this->getRepository();
-        if (null !== $this->property) {
-            $user = $repository->findOneBy([$this->property => $username]);
-        } else {
-            if (!$repository instanceof UserLoaderInterface) {
-                throw new \InvalidArgumentException(sprintf('You must either make the "%s" entity Doctrine Repository ("%s") implement "Symfony\Bridge\Doctrine\Security\User\UserLoaderInterface" or set the "property" option in the corresponding entity provider configuration.', $this->classOrAlias, \get_class($repository)));
-            }
 
-            $user = $repository->loadUserByUsername($username);
+        try {
+            if (null !== $this->property) {
+                $user = $repository->findOneBy([$this->property => $username]);
+            } else {
+                if (!$repository instanceof UserLoaderInterface) {
+                    throw new \InvalidArgumentException(sprintf('You must either make the "%s" entity Doctrine Repository ("%s") implement "Symfony\Bridge\Doctrine\Security\User\UserLoaderInterface" or set the "property" option in the corresponding entity provider configuration.', $this->classOrAlias, \get_class($repository)));
+                }
+
+                $user = $repository->loadUserByUsername($username);
+            }
+        } catch (\Exception $e) {
```

TODO: do not catch all exceptions but a more precise one.